src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIRLoweringPhase.cs

```diff
@@ -134,12 +134,19 @@ public override void VisitImportSpan(AddImportChunkGenerator chunkGenerator, Spa
 
             public override void VisitAddTagHelperSpan(AddTagHelperChunkGenerator chunkGenerator, Span span)
             {
-                _builder.Push(new DirectiveIRNode()
+                var directiveNode = new DirectiveIRNode()
                 {
                     Name = CSharpCodeParser.AddTagHelperDirectiveDescriptor.Directive,
                     Descriptor = CSharpCodeParser.AddTagHelperDirectiveDescriptor,
                     Source = BuildSourceSpanFromNode(span),
-                });
+                };
+
+                for (var i = 0; i < chunkGenerator.Diagnostics.Count; i++)
+                {
+                    directiveNode.Diagnostics.Add(chunkGenerator.Diagnostics[i]);
+                }
+
+                _builder.Push(directiveNode);
```

Comment:
My thought was that errored directive nodes can happen in many different ways:

- Malformed extensible directives. 1 lowering point.
- Malformed built-in directives. These each have their own lowering points.
- After-lowering "malformed" extensible directives. Basically someone has their own logic that determines if a directive is valid/invalid and adds diagnostics to a `DirectiveIRNode` that previously wasn't considered "malformed"

Since there's multiple ways the `DirectiveIRNode`s can flow into the system as malformed I thought it made the most sense to have a single deciding point on if a directive was malformed. 

Figured the only negative of having the pass was in the opt-in customization scenario where a users pass would need to pick an appropriate order in relation to the `MalformedDirectiveIRPass`. That negative also enables users to invalidate directives on their own terms though. I can see it going either way 😄